### DISCUSSION TITLE
Ideas for a dexidp operator

### DISCUSSION THREAD

**Discussion by guerzon** [dexidp/dex#3171] [Ideas]
Hey there,

I am currently starting on a project for a dexidp k8s operator. What would you like to see in it?

Lester

**Comment by erikgb** [comment]
@guerzon Are you still working on the operator or is it done? I wondered about the same, and [asked about it on the dexip Slack channel](https://kubernetes.slack.com/archives/C011URMR41W/p1734210847635199), but no responses so far.

> What would you like to see in it?

For me/us an API for the management of dex clients would be most important.

**Comment by stone-z** [comment]
We (Giant Swarm) have a highly opinionated operator ([dex-operator](https://github.com/giantswarm/dex-operator)) we've built to automate the configuration of dex instances in clusters we manage. It supports mostly "dex fleet management" features like renewing app registrations and setting up default configuration. I doubt it is directly interesting for anyone else, but figured I'd mention it just in case.

**Comment by ssyno** [comment]
Hey @guerzon, @erikgb and @stone-z,

I just wanted to share a bit more about what we at Giant Swarm have built as a dex-operator for our platform needs. While ours is pretty tied to our setup reconciling our custom resources, here are some key technical challenges we tackled:

- Automated identity provider management with credential lifecycle handling by building integrations for Azure AD and GitHub with auto-cred rotation.
- Creation of an extensible architecture for supporting multiple auth providers through a common interface.
- There are some tricky bits around managing callbacks across multiple clusters.

We'd be happy to collaborate on something more general-purpose. 
Could definitely share more about what worked well and what we'd do differently.

**Reply by aescanero** [reply]
Which kind of objects have?
Does it reconcile objects with DEX via GPRC?
Are objects able to use secrets?

In our case, I need to generate "oauth2client" on demand, we are using grpc for this, but we don't like this model and want to use an operator instead.
